pyproject.toml

```diff
@@ -56,7 +56,7 @@ test = [
     "beartype<0.16.0",
     "datasets",
     "responses",
-    "llama-cpp-python",
+    "llama-cpp-python==0.2.37",
```

Comment:
Does this really need to be pinned?

Reply:
0.2.38 seemed to have a bug. Older versions worked well. There was also a bug ticket. I will check if it's solved.

Reply:
So this could be something more fundamental and has nothing to do with the implementation. Maybe something model specific. llama-cpp-python >= 0.2.38 is breaking the generation for e-mail adresses...not sure what this is. Maybe we could just pin it for this PR and investigate further in a new PR.

Reply:
Sounds good to me, we need to open an issue before merging the PR

Reply:
https://github.com/outlines-dev/outlines/issues/638

Reply:
My fix for `llama-cpp-python` is in `v0.2.42`, we should pin accordingly.